### PR TITLE
Support of .NET Core

### PR DESCRIPTION
**Pull request by pieceofsummer**
Minor source adjustments and build scripts to work with .NET Core.

Minimal version of `MongoDB.Driver` is set to `2.3.0-beta1`, and this also satisfies custom builds' suffixes (`2.3.0-local-XXXX`), allowing to replace official beta package with one built from source (for instance, I'm using a custom one because a required PR is not merged yet).
This should be changed to `2.3.0` after it is officially released.

Minimal version of `Hangfire.Core` is bumped to `1.6.0` (as it was the first version targeting .NET Core).

### CODE REVIEW

File: build/build.ps1 [hunk 1]
```diff
@@ -72,7 +76,7 @@ Task InitConfiguration -depends ValidateConfig, GetProjectVersion -description "
 Task CreateOutputDirs -depends InitConfiguration -description "Creates all necessary folders in ""Artifacts"" folder" {
 
 	if (Test-Path $artifacts_path) {
-		Remove-Item -Recurse -Force $artifacts_path;
+		Remove-Item $artifacts_path -Recurse -Force;
 	}
```
Comment (gottscj) [inline build/build.ps1]: Is this change needed to support .NET Core?

File: src/Hangfire.Mongo/CountersAggregator.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace Hangfire.Mongo
     /// </summary>
     public class CountersAggregator : IBackgroundProcess, IServerComponent
     {
-        private static readonly ILog Logger = LogProvider.GetCurrentClassLogger();
+        private static readonly ILog Logger = LogProvider.For<CountersAggregator>();
```
Comment (gottscj) [inline src/Hangfire.Mongo/CountersAggregator.cs]: Is this change needed to support .NET Core?

File: src/Hangfire.Mongo/DistributedLock/MongoDistributedLock.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ namespace Hangfire.Mongo.DistributedLock
     public sealed class MongoDistributedLock : IDisposable
     {
 
-        private static readonly ILog Logger = LogProvider.GetCurrentClassLogger();
+        private static readonly ILog Logger = LogProvider.For<MongoDistributedLock>();
```
Comment (gottscj) [inline src/Hangfire.Mongo/DistributedLock/MongoDistributedLock.cs]: Is this change needed to support .NET Core?

File: src/Hangfire.Mongo/ExpirationManager.cs [hunk 4]
```diff
@@ -9,12 +9,12 @@
...
-    /// </summary>
-    public class ExpirationManager : IBackgroundProcess, IServerComponent
-    {
-        private static readonly ILog Logger = LogProvider.GetCurrentClassLogger();
+	/// <summary>
+	/// Represents Hangfire expiration manager for Mongo database
+	/// </summary>
+	public class ExpirationManager : IBackgroundProcess, IServerComponent
+	{
+		private static readonly ILog Logger = LogProvider.For<ExpirationManager>();
```
Comment (gottscj) [inline src/Hangfire.Mongo/ExpirationManager.cs]: Is this change needed to support .NET Core?

File: src/Hangfire.Mongo/MongoUtils/AutoIncrementIdGenerator.cs [hunk 5]
```diff
@@ -38,7 +38,7 @@ protected AutoIncrementIdGenerator(string prefix)
         /// <returns>Generated identifier</returns>
         public object GenerateId(object container, object document)
         {
-            Type containerType = container.GetType();
+            TypeInfo containerType = container.GetType().GetTypeInfo();
```
Comment (gottscj) [inline src/Hangfire.Mongo/MongoUtils/AutoIncrementIdGenerator.cs]: Is this change needed to support .NET Core?

File: build/build.ps1 (hunk 1 above)
Comment (pieceofsummer) [inline build/build.ps1]: Not really, just was changed at some moment

File: src/Hangfire.Mongo/CountersAggregator.cs (hunk 2 above)
Comment (pieceofsummer) [inline src/Hangfire.Mongo/CountersAggregator.cs]: Yes, there's no `GetCurrentClassLogger()` method in .netstandard version of Hangfire

File: src/Hangfire.Mongo/MongoUtils/AutoIncrementIdGenerator.cs (hunk 5 above)
Comment (pieceofsummer) [inline src/Hangfire.Mongo/MongoUtils/AutoIncrementIdGenerator.cs]: Yes

File: src/Hangfire.Mongo/ExpirationManager.cs (hunk 4 above)
Comment (pieceofsummer) [inline src/Hangfire.Mongo/ExpirationManager.cs]: Same here

**Comment by pieceofsummer** [comment]
@gottscj not sure if I can make it happy. It does some weird things in the AppVeyor environment, which I cannot reproduce locally, so it will be just randomly changing some settings and seeing if it builds.

So it probably will make sense to either migrate to .xproj's once and for all, or to place dotnet project files into separate folders, so MSBuild won't even try to look for them. Your choice.

**Comment by gottscj** [comment]
@pieceofsummer, thank you for your quick reply. Let me get back to you on that.

**Comment by gottscj** [comment]
@pieceofsummer, 
Thank you for your patience. We are planning to target .NetStandard. Would that meet your needs?
https://docs.microsoft.com/en-us/dotnet/articles/standard/library

**Comment by dsoronda** [comment]
Is there any reason why this isn't merged yet ?

**Comment by gottscj** [comment]
Hi.
Thank you for your patience.
We would really like AppVeyor to be happy. We tried to clone the pull request (#39) and could not build it on our local VS either?
We are a little busy at the moment, but we will put some effort into getting this PR working and merget. Thanks again @pieceofsummer for the work you have put in.

**Comment by MarLoe** [comment]
Hi @pieceofsummer 

I have checked out this pull request locally and fixed it, so that it should be running both .net45 AND .netstandard1.5.
I have uploaded the result in a separate branch:
https://github.com/sergeyzwezdin/Hangfire.Mongo/tree/dotnetcore

I hope I have succeeded. Please take a look (perhaps pull and double check). Any comments are welcome.

I plan to do a pull request after the week-end and get into master asap after that.

**Comment by MarLoe** [comment]
Pull Request #49 has been merged into master
